Add-a-spinner-at-least-until-some-messages-are-shown

David-Muhasa opened the pull request:
As part of, or after Bootstrap 5: Cypht: Add a spinner at least until some messages are shown
In lists
For a message
Search
etc.

Shadow243 reviewed (changes requested):
Please make sure all tests pass successfully

Shadow243 reviewed (changes requested):
I like the idea there is just one problem when we are on the compose
 page, cypht is making ajax requests on the hook: ajax_imap_message_content and this is preventing someone from continuing to write until that this request ends, several times.

<img width="1136" alt="Screenshot 2024-07-25 at 18 00 18" src="https://github.com/user-attachments/assets/07bf2f07-45ea-448f-9576-c7a81f0fea08">

Try to find a solution to this.

Shadow243 commented:
https://github.com/cypht-org/cypht/actions/runs/10211842394/job/28253942848?pr=1126

Shadow243 commented:
You changed `loading_icon` to 
```
<div class="loading_bar"></div>\n
   │ +            <div class="loading_spinner text-center">\n
   │ +                <div class="spinner-border" style="width: 3rem; height: 3rem;" role="status">\n
   │ +                    <span class="visually-hidden">Loading...</span>\n
   │ +                </div>\n
   │ +            </div>'
   ```
Test:
 ✘ Loading icon
at:
   `/home/runner/work/cypht/cypht/tests/phpunit/modules/core/output_modules.php:459`

David-Muhasa commented:
I changed loading_icon to  loading_bar

> You changed `loading_icon` to
> 
> ```
> <div class="loading_bar"></div>\n
>    │ +            <div class="loading_spinner text-center">\n
>    │ +                <div class="spinner-border" style="width: 3rem; height: 3rem;" role="status">\n
>    │ +                    <span class="visually-hidden">Loading...</span>\n
>    │ +                </div>\n
>    │ +            </div>'
> ```
> 
> Test: ✘ Loading icon at: `tests/phpunit/modules/core/output_modules.php:459`

I changed loading icon to loading bar to be very specific

Shadow243 commented:
> I changed loading_icon to loading_bar
> 
> > You changed `loading_icon` to
> > ```
> > <div class="loading_bar"></div>\n
> >    │ +            <div class="loading_spinner text-center">\n
> >    │ +                <div class="spinner-border" style="width: 3rem; height: 3rem;" role="status">\n
> >    │ +                    <span class="visually-hidden">Loading...</span>\n
> >    │ +                </div>\n
> >    │ +            </div>'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Test: ✘ Loading icon at: `tests/phpunit/modules/core/output_modules.php:459`
> 
> I changed loading icon to loading bar to be very specific

Okay, you also need to go change this in the unit tests at this line: `tests/phpunit/modules/core/output_modules.php:459`

Shadow243 commented on modules/core/output_modules.php:
```diff
@@ -650,7 +650,13 @@ class Hm_Output_loading_icon extends Hm_Output_Module {
      * Sort of ugly loading icon animated with js/css
      */
     protected function output() {
-        return '<div class="loading_icon"></div>';
```
You renamed the class here: class="loading_bar" but it was refering to the module: `Hm_Output_loading_icon` you should keep it like that.

Shadow243 commented:
I'm trying to think of an example from which we can get an idea: nprogress

Can you do something like that ?

marclaporte commented:
@Shadow243: @David-Muhasa won't have time to finish. Please take over.

josaphatim commented:
After merging #266, I see a loading spinner at the bottom left of the page. Do we still need this?

Shadow243 commented:
> After merging #266, I see a loading spinner at the bottom left of the page. Do we still need this?
> 

I'm also not sure. I'l just confirm and close if that's the case

josaphatim commented:
I think this should only be displayed on the part where content is being loaded to allow users click on links I they want to change a page for example.

Shadow243 commented:
> After merging #266, I see a loading spinner at the bottom left of the page. Do we still need this?

Yes @josaphatim we do , i can see that `loading_icon` div is still there

Shadow243 commented:
@Baraka24 did you assign to someone ?

Baraka24 commented:
> @Baraka24 did you assign to someone ?

No one for now.
